mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -54,6 +54,8 @@ def CExpression : NativeOpTrait<"emitc::CExpression">;
 def IntegerIndexOrOpaqueType : Type<CPred<"emitc::isIntegerIndexOrOpaqueType($_self)">,
 "integer, index or opaque type supported by EmitC">;
 def FloatIntegerIndexOrOpaqueType : AnyTypeOf<[EmitCFloatType, IntegerIndexOrOpaqueType]>;
+def SwitchOperandType : Type<CPred<"emitc::isSwitchOperandType($_self)">,
+"integer type for switch operation">;
```

Comment:
If there is no good reason why pointer wide types should not be used, this should be dropped in favor of using the existing `IntegerIndexOrOpaqueType` which is referring to `isIntegerIndexOrOpaqueType`.

Reply:
We will support them too. Must I add respective test cases?

Reply:
Added tests.